hilo.rb

```diff
@@ -1,34 +1,113 @@
 #!/usr/bin/env ruby
-puts "Welcome to the Hi-Lo game.\nGuess a random whole number from 1 to 100 (inclusive)."
-puts "You will have a total of 6 guesses."
-puts ""
 
-random_num = rand(1..100)
+# Game to guess random number
+class HiLoGame
 
-# gives the user 6 guesses
-for i in 0..5
-    guesses_left = 6 - i
-    puts "You have #{guesses_left} guesses left."
-    puts "Please guess a whole number."
+    # Constructor to initialise the class with a maximum number of guesses
+    def initialize(max_guesses)
+	@max_guesses = max_guesses
+	@guesses_left = @max_guesses
+	@already_guessed = Array.new
+	@random_num = get_random_num
+	welcome
+    end
+
+    def get_random_num
+	random_num = rand(1..100)
+    end
+
+    # Welcome message for the game
+    def welcome
+	puts "Welcome to the Hi-Lo game.\nGuess a random whole number from 1 to 100 (inclusive)."
+	puts "You will have a total of #{@max_guesses} guesses.\nYou cannot guess the same number more than once."
+	puts ""
+    end
+
+    # The game is started
+    def play
+
+	# Gives the user max number of guesses
+	while game_running?
+	    print_guesses_left
+	    guess = guess_a_number
+	    check_guess(guess)
+	    @guesses_left -= 1
+
+	    if has_won?(guess)
+	        print_has_won
+	    	return
+	    end
+	end
+	
+	print_lost_game
+    end
+
+    # Returns true if there are turns left
+    def game_running?
+	@guesses_left > 0
+    end
+
+    # Returns true if guess equals the random number
+    def has_won?(guess)
+	guess == @random_num
+    end
 
-    # only allow integer input
-    guess = gets
-    while guess.strip != guess.to_i.to_s  
-        puts "Guess a whole number!"
-        guess = gets
+    def print_guesses_left
+	puts "You have #{@guesses_left} guesses left.\nPlease guess a whole number from 1 to 100 (inclusive)."
     end
-    guess = guess.to_i
 
-    # comparing randomly generated number with user guess
-    if random_num == guess
+    # Allows user to guess number and validates input by calling validate_guess
+    def guess_a_number	
+	guess = gets
+	guess = validate_guess(guess)
+	@already_guessed.append(guess)
```

Comment:
Not that there's anything wrong with `append` but there is a nice shorthand `@already_guessed << guess`